### ISSUE TITLE
Use custom Jitsi Server

### ISSUE THREAD

**Issue by seppowalther** [mischkadb/lidarts#221]
I EXTREMELY like Lidarts.
Good job, thank you @mischkadb .

I somewhere picked up the idea of using a custom Jitsi-Server instead of the "Lidarts"-Jitsi-Server or the Public Server.
This would have quite a few positive aspects, for example relieving the Lidarts-Jitsi-Server while not necessarily having to use American Servers hosted by AWS, where privacy or latency could become more of an issue.

Had a glance on the code, and it doesn't seem to hard to implement.

**Comment by theduusK** [comment]
Hey,

Just a short info:
The "Lidarts" Jitsi server is not hosted on AWS or any other cloud provider.
I'm hosting the instance on my private dedicated root server, hosted in Düsseldorf, Germany.
And the server is not owned by Lidarts in any way. I'm providing this service as a support for the project, kind of a contribution.

The public server is using the official Jitsi servers.

**Comment by seppowalther** [comment]
Hey @theduusK ,

thank you very much for hosting and providing dusk-server Jitsi instance for the Lidarts community in the first place.

Thanks for your response, even though thats nothing new to me, I did already lookup the IP-adress of the "Lidarts"-Server (even though it doesnt seem to be part of Lidarts itself as you just mentioned).

One reason for creating this issue and "request" was that I already experienced congestion issues during peak hours with the jitsi.dusk-server.de-Server.
Of course you could use the Public Jitsi server, but the "official Jitsi servers" are hosted by AWS and so are hosted in the US.

Another reason is that you seemingly have activated P2P-Sessions on jitsi.dusk-server.de, which is generally speaking a good thing because you save server performance and bandwidth and improve latency a bit, but is in a beta-status and sometimes causes issues when entering with P2P-incompatible Clients or when rejoining, cause P2P only works with 2 participants, so the server has to switch to videobridge-mode as soon as a third participant enters and has to switch back to P2P when the third participant leaves, and every switch causes 0.2 - 2 seconds loss of connection between the participants, depending on the clients and their general ability to operate in P2P-mode.
Switching to Public Jitsi doesnt't help either, P2P is active there as well.

Long story short, we haven't had a good experience with P2P-mode which is one reason why we deactivated this feature on our Jitsi instance.

These are just 2 reasons why it would be handy to have the option to use a self-hosted Jitsi server, because there you would be in control of your own Jitsi instance, own settings, own security protection measures.

But in general, Lidarts is a wonderful peace of software and I enjoy using this platform so much. :)

**Comment by mischkadb** [comment]
There are a few issues with the implementation of private Jitsi servers. I do not see an obvious way to match players to a server both can safely agree to. As admin I cannot trust every user's own instance in terms of privacy and reliability. They'd need individual CSP entries as I do not use wildcards at the moment.

In case of outages or any webcam issues I'm the one who gets all support requests although I barely have control over the Jitsi part. It's already the reason for the majority of requests at the moment and I'm glad that it's just @theduusK I need to ask.

However I agree that the public server and P2P is something that we should get rid of. I only readded the public server as an option because there were reports of issues with our Jitsi service. P2P could be disabled testwise if the bandwidth costs are acceptable.

My prefered solution are small self-hosted instances of Jitsi Meet with a load balancer and disabled P2P for testing purposes. However I am hesitant to try it myself because of the time commitment and costs which might easily double. I cannot estimate the required server resources and bandwith (without P2P) for our current userbase though. We can discuss that if you're interested though.

**Comment by theduusK** [comment]
@mischkadb 
If you want, I can deactivate P2P to see how it affects. 
I think the only reason for using it, was to save some server bandwidth.

**Comment by seppowalther** [comment]
@mischkadb Makes sence, I didn't think of the question how to guarantee a safe and secure Jitsi instance and the support issues.

**Comment by mischkadb** [comment]
We've updated our Jitsi server to the latest version and disabled P2P. Performance does not seem to be an issue at the moment. However we have an option to add more videobridges to scale horizontally.

We might even switch to another WebRTC server soon. Private servers are not an option at the moment though, sorry.
